File: tools/parity/dispatch/wsl-paths.ts
// TS dispatch for the wsl-paths parity module: maps the shared vector function
// names to the real `src/shared/wsl-paths.ts` exports so the harness compares
// the live TS reference against the Rust port.

import { isWslUncPath, parseWslUncPath } from '../../../src/shared/wsl-paths'

export function dispatch(fn: string, input: unknown): unknown {
  switch (fn) {
    case 'parseWslUncPath':
      return parseWslUncPath(input as string)
    case 'isWslUncPath':
      return isWslUncPath(input as string)
    default:
      throw new Error(`unknown function ${fn}`)
  }
}

File: rust/crates/orca-dispatch/src/modules/wsl_paths.rs
//! Parity dispatch for `orca_core::wsl_paths` vs `src/shared/wsl-paths.ts`.

use orca_core::wsl_paths::{is_wsl_unc_path, parse_wsl_unc_path, WslUncPathInfo};
use serde_json::{json, Value};

pub fn dispatch(function: &str, input: &Value) -> Value {
    match function {
        "parseWslUncPath" => match input.as_str() {
            Some(path) => info_to_json(parse_wsl_unc_path(path)),
            // Vectors always pass a string path; a non-string is a vector bug.
            None => json!({ "__parity_error__": "expected string input for parseWslUncPath" }),
        },
        "isWslUncPath" => match input.as_str() {
            Some(path) => Value::Bool(is_wsl_unc_path(path)),
            None => json!({ "__parity_error__": "expected string input for isWslUncPath" }),
        },
        other => json!({ "__parity_error__": format!("unknown function {other}") }),
    }
}

/// Match `JSON.stringify` of the TS `WslUncPathInfo | null` return: `null` when
/// no match, else `{ distro, linuxPath }` with the camelCase TS field names.
fn info_to_json(info: Option<WslUncPathInfo>) -> Value {
    match info {
        Some(info) => json!({
            "distro": info.distro,
            "linuxPath": info.linux_path,
        }),
        None => Value::Null,
    }
}
